src/identity/interaction/HtmlViewHandler.ts

```diff
@@ -0,0 +1,57 @@
+import { BasicRepresentation } from '../../http/representation/BasicRepresentation';
+import type { Representation } from '../../http/representation/Representation';
+import { cleanPreferences, getTypeWeight } from '../../storage/conversion/ConversionUtil';
+import { APPLICATION_JSON, TEXT_HTML } from '../../util/ContentTypes';
+import { MethodNotAllowedHttpError } from '../../util/errors/MethodNotAllowedHttpError';
+import { NotFoundHttpError } from '../../util/errors/NotFoundHttpError';
+import { NotImplementedHttpError } from '../../util/errors/NotImplementedHttpError';
+import type { TemplateEngine } from '../../util/templates/TemplateEngine';
+import type { InteractionHandlerInput } from './InteractionHandler';
+import { InteractionHandler } from './InteractionHandler';
+import type { InteractionRoute } from './routing/InteractionRoute';
+
+/**
+ * Stores the HTML templates associated with specific InteractionRoutes.
+ * Template keys should be file paths to the templates,
+ * values should be the corresponding routes.
+ *
+ * Will only handle GET operations for which there is a matching template if HTML is more preferred than JSON.
+ *
+ * Templates will receive the parameter `idpIndex` in their context pointing to the root index URL of the IDP API.
+ */
+export class HtmlViewHandler extends InteractionHandler {
+  private readonly idpIndex: string;
+  private readonly templateEngine: TemplateEngine;
+  private readonly templates: Record<string, string>;
+
+  public constructor(index: InteractionRoute, templateEngine: TemplateEngine,
+    templates: Record<string, InteractionRoute>) {
+    super();
+    this.idpIndex = index.getPath();
+    this.templateEngine = templateEngine;
+    this.templates = Object.fromEntries(
+      Object.entries(templates).map(([ template, route ]): [ string, string ] => [ route.getPath(), template ]),
+    );
+  }
+
+  public async canHandle({ operation }: InteractionHandlerInput): Promise<void> {
+    if (operation.method !== 'GET') {
+      throw new MethodNotAllowedHttpError();
+    }
+    if (!this.templates[operation.target.path]) {
+      throw new NotFoundHttpError();
+    }
+    const preferences = cleanPreferences(operation.preferences.type);
+    const htmlWeight = getTypeWeight(TEXT_HTML, preferences);
+    const jsonWeight = getTypeWeight(APPLICATION_JSON, preferences);
+    if (jsonWeight >= htmlWeight) {
+      throw new NotImplementedHttpError('HTML views are only returned when they are preferred.');
+    }
```

Comment:
So no full conneg? Seems a bit weird to reinvent.

Reply:
I'm not saying we should do full conneg, but rather that this class shouldn't be doing any conneg IMHO.

Reply:
Would have to think about possible alternatives when going through this PR again. Problem is that we only want to return this in case HTML is the most preferred (similar to how we handle index resources). 

We could do conneg by creating a converter that checks the target identifier and knows which HTML page to return, but that would give weird results if for some reason someone does a POST with `accept: text/html` since that would execute the API action but would then return the input page again.

Could also wrap this class in a small class that does this semi-conneg thing.

Reply:
Good for now; maybe just leave a TODO.